src/include/duckdb/function/table/arrow.hpp

```diff
@@ -25,6 +25,8 @@ struct ArrowScanFunctionData : public TableFunctionData {
 	unique_ptr<ArrowArrayStreamWrapper> stream;
 	//! This holds the original list type (col_idx, [ArrowListType,size])
 	std::unordered_map<idx_t, vector<std::pair<ArrowListType, idx_t>>> original_list_type;
+	//! This holds the index type for dictionaries
+	std::unordered_map<idx_t, LogicalType> dictionary_type;
```

Comment:
This seems to assume there is one dictionary only and that it is the root element, what if there is a struct vector that holds dictionaries? Or a list vector that holds a dictionary? Can that happen?

Reply:
Excellent point, that was indeed my original assumption.

Reply:
```python
indices_list = [ 0, 1, 2, 3, 4]
indices = pa.array(indices_list * 1000,pyarrow.int64())
dictionary = pa.array([1,2,3,4,5],pyarrow.int64())
dict_array = pa.DictionaryArray.from_arrays(indices, dictionary)
data = pyarrow.array([[dict_array],None, [dict_array]], type=pyarrow.large_list(dict_array.type))
```

pyarrow.lib.ArrowInvalid: Could not convert <pyarrow.lib.DictionaryArray object at 0x7ffae48553c0>


It seems my original assumption was correct :)